### ISSUE TITLE
When Jetpack REST is enabled, isWPcom isn't accurate anymore

### ISSUE THREAD

**Issue by koke** [wordpress-mobile/WordPress-iOS#3700]
Since Jetpack REST will migrate Jetpack blogs to blogs under the .com account, `isWPcom` will be `YES`.

We need to evaluate `isWPcom` usage across the app and come up with alternatives:
- sometimes, when a piece of code asks `isWPcom` it really wants to know if the site is hosted in wp.com to enable a specific feature
- sometimes those features are also available in Jetpack
- sometimes the code really wants to ask if the blog supports REST

**Comment by koke** [comment]
A list of things that `isWPcom` is used for (there are a few more cases that I haven't included since I though that code shouldn't even be checking `isWPcom`):

- Which default blavatar to use: can we use the same for all blogs?
- is private: a wp.com only feature
- enable REST: used by `Blog` internally to figure out if REST should be enabled
- "Remove Site" enabled: only for self hosted, shouldn't be available for Jetpack REST either
- "Hide blog" enabled: reverse of the previous one, only blogs that aren't from the default wp.com account should be allowed to hide/show
- VideoPress enabled: I think this should be `YES` for everyone
- `lastUsedOrFirstWPcomBlog` which is really used to get the default blog for reblogging (but the blog selector doesn't show .com blogs only): I think it's a wp.com feature only but I'd have to check
- can use a native reader view for posts or has to load a web view: it might work for Jetpack sites, I'd have to check, @aerych @oguzkocer ?
- likes are enabled: should be available for both wp.com and Jetpack sites (for own blogs)
- comments/likes available for post cards: should be available for both wp.com and Jetpack sites
- stats for a specific post are available: I'd think that would work on a Jetpack site but I'd have to check, @astralbodies ?
- uses OAuth2 for wp-login hack: it's more of a has token/has password question
- supports mentions (for SuggestionsService): wp.com only AFAIK, @allendav ?
- private URL protocol for post views on private sites: wp.com only feature, although we could be more liberal when deciding when to enable it as long as we only authenticate wp.com requests

I'm leaning towards adding a `-[Blog supports:]` method and a bunch of feature constants for each of those cases. Then, even if we mostly use that we should differentiate between `isHostedByWPCOM` and `usesWPCOMRESTAPI, which are probably the ugliest method names ever.

**Comment by koke** [comment]
> Which default blavatar to use: can we use the same for all blogs?

calypso uses the genericon for website for both .com and Jetpack sites, and it's generic enough that we could use it as well for self hosted. I'm not sure who to ping here, so I'll go with @mnt, feel free to point me to someone else :smile: 

![screen shot 2015-05-21 at 00 45 07](https://cloud.githubusercontent.com/assets/8739/7738257/abd2fa3a-ff52-11e4-86bd-fe4966412cde.png)

**Comment by koke** [comment]
> I'm leaning towards adding a -[Blog supports:] method and a bunch of feature constants for each of those cases.

I did a first pass in 898cc3b. I'm not happy about calling `AccountService` from `Blog` but other than that I'm liking it. 

@sendhil and everyone else I pinged before, does this approach look good?

**Comment by koke** [comment]
website noticon looks good to me, yay vector graphics! 2d3d168

![ios simulator screen shot 21 may 2015 11 35 42](https://cloud.githubusercontent.com/assets/8739/7745716/95101d4a-ffad-11e4-9062-045157e8aeb3.png)

**Comment by koke** [comment]
I missed another one:
- comment likes: I think it's a wp.com only feature and I treated it as such in ea248be

**Comment by koke** [comment]
Another one:
- can we toggle push notifications for the blog? I'd say YES if `blog.account` is the default wp.com account. I don't think push would work correctly for Jetpack sites connected to a non-default account. @jleandroperez thoughts?

**Comment by koke** [comment]
@astralbodies there are a bunch of `isWPcom` checks in `StatsWebViewController`. Since there's no way to get to the web stats unless native stats are working, can we drop all the checks in there?

**Comment by koke** [comment]
I merged #3689 into `issue/3700-fix-iswpcom` to avoid updating `Blog+Jetpack`

**Comment by astralbodies** [comment]
@koke I believe we can. That VC is rather rusty so it could use some cleanup. You want that done in your PRs or should I open a PR against this issue?

**Comment by koke** [comment]
A separate PR sounds good.

**Comment by jleandroperez** [comment]
@koke i'm pretty sure you're right, if the blog is connected to a secondary wpcom account, it should work (AFAIK!)

**Comment by oguzkocer** [comment]
> calypso uses the genericon for website for both .com and Jetpack sites, and it's generic enough that we could use it as well for self hosted. I'm not sure who to ping here, so I'll go with @mnt, feel free to point me to someone else :smile:

I've made a similar change for Android and it's a good idea to make the change in iOS as well. The screenshot looks good to me, but as you said, @mnt would be the right person for this.

> can use a native reader view for posts or has to load a web view: it might work for Jetpack sites, I'd have to check

I'll defer to @aerych for this.

> supports mentions (for SuggestionsService): wp.com only AFAIK?

@allendav is afk this week and you're right it's wp.com only feature.

**Comment by sendhil** [comment]
```
I did a first pass in 898cc3b. I'm not happy about calling AccountService from Blog but other than that I'm liking it.

@sendhil and everyone else I pinged before, does this approach look good?
```

@koke - approach looks good to me..

**Comment by aerych** [comment]
> can use a native reader view for posts or has to load a web view: it might work for Jetpack sites, I'd have to check

I'm not sure what this is about.

**Comment by koke** [comment]
@aerych this c416171543e3eadd545cdf924fbc49fb09c7c0a2

Can ReaderPostDetailViewController display a post from a Jetpack site, as long as it's using REST?

**Comment by aerych** [comment]
> Can ReaderPostDetailViewController display a post from a Jetpack site, as long as it's using REST?

Yeppers!

**Comment by koke** [comment]
Waiting on #3689 to submit the PR
Progress so far: https://github.com/wordpress-mobile/WordPress-iOS/compare/issue/3700-fix-iswpcom

**Comment by mnt** [comment]
> calypso uses the genericon for website for both .com and Jetpack sites, and it's generic enough that we could use it as well for self hosted. I'm not sure who to ping here, so I'll go with @mnt, feel free to point me to someone else :smile:

Whoops, just noticed this—looks perfect though, you didn't need me. :)

**Comment by allendav** [comment]
@koke wrote:

> supports mentions (for SuggestionsService): wp.com only AFAIK, @allendav ?

Sorry - missed that while I was out - yes that's correct - mentions are not supported on Jetpack sites - they are for WPCOM sites only.
